src/pagination.rs

```diff
@@ -0,0 +1,264 @@
+    pub entries_per_page_choices: Vec<i32>,
+    #[prop_or(20)]
+    pub selected_choice: i32,
+
+    // callback for the buttons
+    #[prop_or_default]
+    pub navigation_callback: Callback<Navigation>,
+
+    #[prop_or_default]
+    pub limit_callback: Callback<i32>,
+}
+
+pub enum Navigation {
+    First,
+    Previous,
+    Next,
+    Last,
+    Page(i32),
+}
+
+pub struct Pagination {
+    expanded: bool,
+    global_close: GlobalClose,
+}
+
+#[derive(Clone, Debug)]
+pub enum Msg {
+    ToggleMenu,
+    CloseMenu,
+    SetLimit(i32),
+
+    First,
+    Previous,
+    Next,
+    Last,
+    Page(String),
+}
+
+impl Component for Pagination {
+    type Message = Msg;
+    type Properties = Props;
+
+    fn create(ctx: &Context<Self>) -> Self {
+        Self {
+            expanded: false,
+            global_close: GlobalClose::new(
+                NodeRef::default(),
+                ctx.link().callback(|_| Msg::CloseMenu),
+            ),
+        }
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::ToggleMenu => {
+                self.expanded = !self.expanded;
+            }
+            Msg::SetLimit(limit) => {
+                ctx.props().limit_callback.emit(limit);
+                ctx.link().send_message(Msg::CloseMenu)
+            }
+            Msg::CloseMenu => self.expanded = false,
+            Msg::First => ctx.props().navigation_callback.emit(Navigation::First),
+            Msg::Previous => ctx.props().navigation_callback.emit(Navigation::Previous),
+            Msg::Next => ctx.props().navigation_callback.emit(Navigation::Next),
+            Msg::Last => ctx.props().navigation_callback.emit(Navigation::Last),
+            Msg::Page(num) => ctx
+                .props()
+                .navigation_callback
+                .emit(Navigation::Page(num.parse::<i32>().unwrap_or(1))),
+        }
+        true
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        // The pagination menu : "1-20 of nnn"
+        let mut menu_classes = Classes::from("pf-c-options-menu");
+        if self.expanded {
+            menu_classes.push("pf-m-expanded");
+        }
+
+        // The default rust div operator does floor(), we need ceil, so we cast to float before doing the operation
+        let max_page = ctx
+            .props()
+            .total_entries
+            .map(|m| (m as f32 / ctx.props().selected_choice as f32).ceil() as i32);
+        let current_page =
+            (ctx.props().offset as f32 / ctx.props().selected_choice as f32).ceil() as i32;
+
+        let is_last_page = if let Some(max) = ctx.props().total_entries {
+            ctx.props().offset + ctx.props().selected_choice > max
+        } else {
+            false
+        };
+
+        let total_entries = ctx
+            .props()
+            .total_entries
+            .map(|m| format!("{}", m))
+            .unwrap_or(String::from("unknown"));
```

Comment:
I would use `unwrap_or_else` to save the function call and allocation if a value is present.